### PR TITLE
OADP-1983 - AWS settings to install OADP for ROSA STS

### PR DESCRIPTION
**Pull request by HagayVider1**
BZ# : 
OCPBUGS# : [OADP-1983](https://issues.redhat.com//browse/OADP-1983)

Version(s): OADP 1.2.1

Issue:  https://issues.redhat.com/browse/OADP-1983

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 1]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+export SCRATCH="/tmp/${CLUSTER_NAME}/oadp"
+mkdir -p ${SCRATCH}
+echo "Cluster ID: ${ROSA_CLUSTER_ID}, Region: ${REGION}, OIDC Endpoint: ${OIDC_ENDPOINT}, AWS Account ID: ${AWS_ACCOUNT_ID}"
+----
+
+. On the AWS account, create an IAM Policy to allow access to S3.
++
+[source, terminal]
+----
+POLICY_ARN=$(aws iam list-policies --query "Policies[?PolicyName=='RosaOadp'].{ARN:Arn}" -- output text)
```
Comment (amastbau) [inline modules/oadp-installing-oadp-rosa-sts.adoc:50]: maybe put the policy name RosaOadp  as doc parameters?

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 2]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "s3:CreateTags",
+      "s3:CreateVolume",
+      "s3:CreateSnapshot",
+      "s3:DeleteSnapshot",
+    ]
+    "Resource": "*"
+  }
+ ]}
+EOF
+POLICY_ARN=$(aws iam create-policy --policy-mane "RosaOadp" \
```
Comment (amastbau) [inline modules/oadp-installing-oadp-rosa-sts.adoc:85]: Maybe add remarks  
(For Example: creating the policy and getting the policy Amazon Resource Name)

**Review by amastbau — COMMENTED** [review]
Maybe add references to aws for more information about Roles, Policies, etc.

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 3]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+[NOTE]
+====
+Change the cluster name to match your ROSA cluster and ensure you’re logged into the cluster as an Administrator. Ensure all fields are outputted correctly before moving on.
+====
++
+[source, terminal]
+----
+export CLUSTER_NAME=my-cluster
+export ROSA_CLUSTER_ID-$(rosa describe cluster -c ${CLUSTER_NAME} --output json | jq -r .id)
+export REGION=$(risa describe cluster -c ${CLUSTER_NAME} --output json | jq -r .region.id)
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:35]: risa -> rosa

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 4]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
++
+[NOTE]
+====
+Change the cluster name to match your ROSA cluster and ensure you’re logged into the cluster as an Administrator. Ensure all fields are outputted correctly before moving on.
+====
++
+[source, terminal]
+----
+export CLUSTER_NAME=my-cluster
+export ROSA_CLUSTER_ID-$(rosa describe cluster -c ${CLUSTER_NAME} --output json | jq -r .id)
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:34]: hyphen after ROSA_CLUSTER_ID to be replaced by =

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 5]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+export REGION=$(risa describe cluster -c ${CLUSTER_NAME} --output json | jq -r .region.id)
+export OIDC_ENDPOINT=$(oc get authentication.config.openshift.io cluster -o jsonpath='{.spec.serviceAccountIssuer}' | sed 's|^https://||')
+export AWS_ACCOUNT_ID='aws sts get-caller-identity --query Account --output text'
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:37]: export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)

File: modules/oadp-installing-oadp-rosa-sts.adoc (hunk 2 above)
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:85]: --policy-mane => --policy-name

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 6]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "s3:GetBucketTegging",
+      "s3:PutEncryptionConfiguration",
+      "s3:GetEncryptionConfiguration",
+      "s3:PutLifecycleConfiguration",
+      "s3:GetLifecycleConfiguration",
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteOgject",
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:71]: DeleteObject

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 7]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+cat << EOF > ${SCRATCH}/policy.json
+{
+"Version": "2012-10-17",
+"Statement": [
+  {
+    "Effect": "Allow",
+    "Action": [
+      "s3:CreateBucket",
+      "s3:DeleteBucket",
+      "s3:PutBucketTegging",
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:61]: Tagging

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 8]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "s3:GetBucketTegging",
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:62]: Tagging

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 9]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "s3:PutEncryptionConfiguration",
+      "s3:GetEncryptionConfiguration",
+      "s3:PutLifecycleConfiguration",
+      "s3:GetLifecycleConfiguration",
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteOgject",
+      "s3:ListBucketMultipartUpLoads",
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:72]: UpLoads -> Uploads

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 10]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "s3:AbortMultipartUpLoads",
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:73]: AbortMultipartUpLoads -> AbortMultipartUpload

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 11]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "s3:ListMultipartUpLoadParts",
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:74]: ListMultipartUpLoadParts -> ListMultipartUploadParts

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 12]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "s3:DescribeSnapshots",
+      "s3:CreateTags",
+      "s3:CreateVolume",
+      "s3:CreateSnapshot",
+      "s3:DeleteSnapshot",
+    ]
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:80]: comma after bracket

Comment (stillalearner) [reply modules/oadp-installing-oadp-rosa-sts.adoc:80]: Pasting the full correct policy here as well:

{
"Version": "2012-10-17",
"Statement": [
  {
    "Effect": "Allow",
    "Action": [
      "s3:CreateBucket",
      "s3:DeleteBucket",
      "s3:PutBucketTagging",
      "s3:GetBucketTagging",
      "s3:PutEncryptionConfiguration",
      "s3:GetEncryptionConfiguration",
      "s3:PutLifecycleConfiguration",
      "s3:GetLifecycleConfiguration",
      "s3:GetBucketLocation",
      "s3:ListBucket",
      "s3:GetObject",
      "s3:PutObject",
      "s3:DeleteObject",
      "s3:ListBucketMultipartUploads",
      "s3:AbortMultipartUpload",
      "s3:ListMultipartUploadParts",
      "s3:DescribeSnapshots",
      "s3:CreateTags",
      "s3:CreateVolume",
      "s3:CreateSnapshot",
      "s3:DeleteSnapshot"
    ],
    "Resource": "*"
  }
 ]}

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 13]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+}
+EOF
+ROLE_ARN=$(aws iam create-role --role-name \
+  "${ROLE_NAME}" \
+  --assume-role-policy-document file://${SCRATCH}/trust-policy.json \
+  --tags Key+rosa_cluster_id,Value=${ROSA_CLUSTER_ID}
+Key=rosa_openshift_verson,Value=${CLUSTER_VERSION}
+Key=rosa_role_prefix,Value=ManagedOpenShift
+Key=operator_namespace,Value=openshift-adp
+Key=operator_name,Value-openshift-oadp \
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:123]: Value=

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 14]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+        }
+      }
+    }]
+}
+EOF
+ROLE_ARN=$(aws iam create-role --role-name \
+  "${ROLE_NAME}" \
+  --assume-role-policy-document file://${SCRATCH}/trust-policy.json \
+  --tags Key+rosa_cluster_id,Value=${ROSA_CLUSTER_ID}
+Key=rosa_openshift_verson,Value=${CLUSTER_VERSION}
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:120]: version

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 15]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_ENDPOINT}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${OIDC_ENDPOINT}:sub": [
+            "system:serviceaccount:openshift-adp:openshift-adp-controller-manager",
+            "system:serviceaccount:openshift-adp:velero:]
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:110]: velero:  -> velero"

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 16]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+echo ${POLICY_ARN}
+----
+
+. Create an IAM Role trust policy for the cluster.
++
+[source, terminal]
+----
+cat <<EOF > ${SCRATCH}/trust-policy.json
+{
+    "Version": :2012-10-17",
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:99]: :2012 -->  "2012

File: modules/oadp-installing-oadp-rosa-sts.adoc [hunk 17]
```diff
@@ -15,9 +15,128 @@ Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the
...
+. Attach the IAM Policy to the IAM Role.
+
++
+[source, terminal]
+----
+aws iam attach-role-policy --role-name "${ROLE_NAME}" \
+  --policy-arn ${POLICY_ARN}
+----
+
+. Continue to install OADP on the cluster.
```
Comment (stillalearner) [inline modules/oadp-installing-oadp-rosa-sts.adoc:138]: There are multiple small errors in this yaml and role command. Please copy exactly from here:

https://mobb.ninja/docs/misc/oadp/rosa-sts/

**Comment by akarol** [comment]
@anarnold97 can we have the commands and the yaml rightly copied?

**Comment by HagayVider1** [comment]
Changes to this pull request were implemented in OSDOCS-6598
https://github.com/openshift/openshift-docs/pull/62016

**Pull request closed without merging**